### PR TITLE
Add templates for final tests for aggregation and fragmentation

### PR DESCRIPTION
**Pull request by WFlynnZ**
This PR aims to implement the tests for aggregation and fragmentation equations. The template will be updated once CADET-Core figures out a neat way to implement all possible situations.

### CODE REVIEW

**Comment by jbreue16** [comment]
Thanks Flynn !
Do we also have a case where aggregation + breakage + DPFR transport happens? Or was that not analyzed in the paper or did I oversee it in the script here or is there something speaking against considering this case?

**Comment by WFlynnZ** [comment]
Hi Jan, we actually have aggregation+breakage+DPFR in our paper, but I did not include it here because I thought we have 1. aggregation and breakage, separately, 2. aggregation and breakage combined, 3. aggregation+DPFR and breakage+DPFR, separately, if these tests all pass, especially we have 2 and 3 passed, what could go wrong if we do aggregation+breakage+DPFR, seems somewhat redundant to me. But I can certainly add if you think it is necessary.

**Comment by jbreue16** [comment]
Hm I see.. I think it would be best to have tests for all combinations listed in [your comment](https://github.com/cadet/CADET-Core/pull/241#issuecomment-2395660106) on the core side. Theres always things that can go wrong with implementation unexpectedly

File: src/test_cadet_core/settings_crystallization.py [hunk 1]
```diff
@@ -0,0 +1,1340 @@
...
+    nComp = n_x + 2
+
+    # crystal space
+    x_grid, x_ct = get_log_space(n_x, x_c, x_max)
+
+    # Boundary conditions
+    boundary_c = nComp*[0.0, ]
+
+    # Initial conditions
+    initial_c = np.asarray([3.0*x_ct[k]**2 * 4.0*x_ct[k]**3 * np.exp(-2.0*x_ct[k]**3)
```
Comment (jbreue16) [inline src/test_cadet_core/settings_crystallization.py:729]: Ive added these initial conditions (copied the aggregation_fragmentation test case), because otherwise nothing happens in this simulation with initial and boundary conditions being zero

**Review by jbreue16 — CHANGES_REQUESTED** [review]
I get weird convergence rates for the PBM_aggregation_fragmentation [(see CSTR_PBM_aggregation_fragmentation.json)](https://github.com/user-attachments/files/20027128/CSTR_PBM_aggregation_fragmentation.json)
image

But for the aggregation_fragmentation test and the pbm test they look as expected ..
Why could that be, is there a mistake in the setup of the model, a difference in the test case compared to the paper?

**Comment by WFlynnZ** [comment]
# Boundary conditions
    boundary_c = []
    for p in range(n_x):
        if p == 0:
            boundary_c.append(c_feed)
        elif p == n_x-1:
            boundary_c.append(c_eq)
        else:
            boundary_c.append(0.0)
    boundary_c = np.asarray(boundary_c)

    # Initial conditions
    initial_c = []
    for k in range(n_x):
        if k == 0:
            initial_c.append(0.0)
        elif k == n_x-1:
            initial_c.append(c_eq)
        else:
            initial_c.append(0.0)
    initial_c = np.asarray(initial_c)

Updated IC and BC.
